### PR TITLE
fix(stream): fix wrong usage of chunk builder in topn

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
fix https://github.com/risingwavelabs/risingwave/issues/5285

### CODE REVIEW

File: src/stream/src/executor/group_top_n.rs [hunk 1]
```diff
@@ -204,7 +204,7 @@ impl<S: StateStore> TopNExecutorBase for InnerGroupTopNExecutorNew<S> {
                 )
                 .await?;
         }
-        // compare the those two ranges and emit the differantial result
```
Comment (xxchan) [inline src/stream/src/executor/group_top_n.rs:207]: The comment is confusing

**Review by st1page — APPROVED** [review]
LGTM

File: src/stream/src/executor/group_top_n.rs (hunk 1 above)
Comment (st1page) [reply src/stream/src/executor/group_top_n.rs:207]: maybe it should be the comment of the function `generate_output`

**Review by st1page — APPROVED** [review]
LGTM to merge it as a fix. but in future, we should try to limit the chunk size in the stream executor like stream hashjoin.

**Pull request merged**